main.tf

```diff
@@ -220,3 +220,9 @@ resource "google_storage_bucket_object" "folders" {
   name     = "${each.value.folder}/" # Declaring an object with a trailing '/' creates a directory
   content  = "foo"                   # Note that the content string isn't actually used, but is only there since the resource requires it
 }
+
+resource "google_storage_hmac_key" "hmac_keys" {
+  for_each              = var.set_hmac_access ? var.hmac_service_accounts : {}
```

Comment:
Can we use var.hmac_service_accounts directly instead of having a bool flag?

Reply:
Sorry, I'm not sure that I understand you clearly. Do you want to use it like this?
`for_each = var.hmac_service_accounts`

Reply:
Right since it's {} by default.

Reply:
Without this condition, we can't drop all hmac access using only one variable "set_hmac_access". 
For example, I have a few service accounts with hmac access, and I want to drop them all temporarily. Maybe in the future, I will want to back. And for more convenient usage I can set the variable to false to delete all hmac access without deleting all lists of service accounts from a code.
I think with this condition it is better.